src/components/ui/BaseCheck.vue

```diff
@@ -1,17 +1,15 @@
 <template>
-  <img
-    :src="currentSvg"
-    class="sm:hidden lg:block lg:pr-4"
-    @click="clickedSvg"
-  />
+  <div @click.stop="clickedSvg">
+    <img :src="currentSvg" @click="checkedTodo" class="pr-3 mt-5" />
+  </div>
 </template>
 
 <script setup lang="ts">
 import { ref } from "vue";
 import ellipseSvg from "../../assets/svg/ellipse.svg";
 import ellipseCheckedSvg from "../../assets/svg/ellipse-checked.svg";
 
-const emit = defineEmits(["update:modelValue"]);
+const emit = defineEmits(["update:modelValue", "checked-todo"]);
```

Comment:
If you created a general component for Check, you shouldn't add the `todo` here in order not to change this to a specific component for Todos. 
In this case you can add everything as general as you can.
 For example:
 Change the `checked-todo` to `on-checked`
 The action should be handled here which is the checked, you don't need to specify what is checked.